File: app/src/main/java/com/botijasoftware/utils/collision/CollisionVolume.java
package com.botijasoftware.utils.collision;

import com.botijasoftware.utils.Vector3;

public class CollisionVolume {
	public boolean collides(CollisionVolume c) {
		return false;
	}
	
	public boolean collides(Vector3 v) {
		return false;
	}
	
	public boolean collides(Sphere s) {
		return false;
	}
	
	public boolean collides(Capsule c) {
		return false;
	}
	
	public boolean collides(Cube c) {
		return false;
	}
	
	public boolean collides(Segment s) {
		return false;
	}
	
	public boolean collides(Ray r) {
		return false;
	}
	
	public boolean collides(Plane p) {
		return false;
	}
	
	public void setPosition(Vector3 v) {
		
	}
	
	public Vector3 getPosition() {
		return new Vector3(0,0,0);
	}
	
	public void setRadius(float r) {
		
	}
	
	public float getRadius() {
		return 0.0f;
	}
}

File: app/src/main/java/com/botijasoftware/utils/collision/CollisionVolume.kt
package com.botijasoftware.utils.collision

import com.botijasoftware.utils.Vector3

open class CollisionVolume {
    open fun collides(c: CollisionVolume): Boolean {
        return false
    }

    open fun collides(v: Vector3): Boolean {
        return false
    }

    open fun collides(s: Sphere): Boolean {
        return false
    }

    open fun collides(c: Capsule): Boolean {
        return false
    }

    fun collides(c: Cube): Boolean {
        return false
    }

    fun collides(s: Segment): Boolean {
        return false
    }

    open fun collides(r: Ray): Boolean {
        return false
    }

    fun collides(p: Plane): Boolean {
        return false
    }

    open var position: Vector3
        get() = Vector3(0f, 0f, 0f)
        set(v) {

        }

    open var radius: Float
        get() = 0.0f
        set(r) {

        }
}